manager/controllers/default/resource/create.class.php

```diff
@@ -188,7 +188,23 @@ public function process(array $scriptProperties = array()) {
      * @return int
      */
     public function getDefaultTemplate() {
-        $defaultTemplate = (isset($this->scriptProperties['template']) ? $this->scriptProperties['template'] : (!empty($this->parent->id) ? $this->parent->get('template') : $this->context->getOption('default_template', 0, $this->modx->_userConfig)));
+        $defaultTemplate = $this->context->getOption('default_template', 0, $this->modx->_userConfig);
+        if (isset($this->scriptProperties['template'])) {
+            $defaultTemplate = $this->scriptProperties['template'];
+        } elseif (!empty($this->parent->id)) {
+            $defaultTemplate = $this->parent->get('template');
+
+            $parentTplObj = $this->modx->getObject('modTemplate', $this->parent->get('template'));
+            if ($parentTplObj) {
+                $props = $parentTplObj->getProperties();
+
+                $childTpl = !empty($props) ? $this->modx->fromJSON($props['childTemplate']) : null;
```

Comment:
Is `$this->modx->fromJSON($props['childTemplate'])` right? Shouldn't it be `$this->modx->fromJSON($props)`?

Reply:
@Mark-H It returns the childTemplate if it is set, so it's a quicker way to retrieve the setting directly

Reply:
@sdrenth Actually no. If `$props['childTemplate']` contains string or integer it works because it is a valid json entity. But there are may happen errors when `childTemplate` not found in `$props`.
Safe variant would be:
```
$props = $parentTplObj->getProperties();

$childTpl = !empty($props) && isset($props['childTemplate']) 
    ? $props['childTemplate'] 
    : null;
```
Also, additionally childTemplate value can be forced to integer by `intval($props['childTemplate'])`